routes/api/posts.js

```diff
@@ -187,78 +189,57 @@ router.post(
   
       Post.findById(req.params.id)
         .then((post) => {
-          if(post){
             const newComment = {
               commentBody: req.body.commentBody,
-              commentedById: req.user.id,
-              commentedByUsername: req.user.username,
-              commentedByAvatar: req.user.avatar
+              username: req.body.username,
+              avatar: req.body.avatar,
+              user: req.user.id,
             };
-            post.comments.unshift(newComment);  // Add to comments array
-            post.save().then((post) => res.json(post)); // Save
-          }else{
-            return res.status(404).json({ success:false, postnotfound: "No post found" });
 
-          }
-        })
-        .catch((err) => res.status(500).json({ success:false, error: err.message }));
+            // Add to comments array
+            post.comments.unshift(newComment);
+
+            // Save
+            post.save().then((post) => res.json(post));
+          })
+            
+        .catch((err) => res.status(404).json({ postnotfound: "No post found" }));
     }
   );
-  
+
   // @route   DELETE api/posts/comment/:id/:comment_id
-  // @desc    Remove comment from post
-  // @input   postId and comment_id from request params
-  // @access  Private
-  router.delete(
-    "/comment/:postId/:commentId",
-    passport.authenticate("jwt", { session: false }),
-    (req, res) => {
+// @desc    Remove comment from post
+// @access  Private
+router.delete(
+  "/comment/:id/:comment_id",
+  passport.authenticate("jwt", { session: false }),
+  (req, res) => {
+    Post.findById(req.params.id)
+      .then((post) => {
```

Comment:
we need to check if `post` is found

Reply:
I think that's what Post.findById(req.params.id) is doing, delete comment works the way it's set up now, when I changed it back to how we originally had the api delete comment wasn't working